ouroboros-consensus/docs/VersioningSchemeDecision.md

```diff
@@ -0,0 +1,577 @@
+
+This rule is so simple that the ✗ for EasyPR is somewhat surpising.
+The hidden problem is that this scheme causes spurious merge conflicts among all your PRs.
+It would only be possible to merge PRs sequentially and merging one PR requires rebasing every other PR and updating its version number diff (assuming a single target branch).
+That usually implies a very poor contributor experience.
+(If you're wondering about variations on this, such as "only bump the version if it hasn't already been bumped", see [below](#RisingEdgeCompromises).)
+
+We assign ✓ for EasyRelease because each release doesn't require any additional work; merely announce the result of the latest PR.
+Relatedly, though, we assign ✗ for TypicalReleases because it's unrealistic to release after every PR.
+Therefore the eventual timeline of your releases will have confusing gaps between the released version numbers.
+
+The opposite scheme would be the most extreme FallingEdge scheme, in which normal PRs never change version declarations, only release-preparation PRs do.
+
+| TypicalReleases | EasyPR | EasyRelease |
+|:---------------:|:------:|:-----------:|
+| ✓               | ✓      | ✗           |
+
+We assign ✓ for TypicalRelease because this scheme constrains neither how often you release nor the evolution of version numbers.
+We similarly assign ✓ for EasyPR because the scheme requires nothing from normal PRs.
+
+However, we assign ✗ for EasyRelease because the scheme requires assessing all of the changes on this branch since the previous release from this branch.
+This is a problem because PRs should be assessed while they're being developed and reviewed, active in the teams' minds, instead of some days/weeks later when the team has "paged out" the details of the PR.
+
+It is unlikely that the version control history, the list of pull requests, or any other _automatic_ log can be used to trivialize the obligation of the post hoc assessment.
+Even if we assume an ideal commit messages and/or PR descriptions, there will likely be a burdensome number of them.
+The slightly more realistic assumptions of commit messages and/or PR descriptions that do have ideal content but don't have perfectly uniform structure means the assessor would still have to thoroughly read each, judging what the content requires of the version bump under consideration.
+
+However, there is one standard log that can often make this assessment much easier: the changelog.
+It's not automatic, but it is often structured around "additions/changes/etc", which are exactly what determines the next version under a policy like SemVer, for example.
+
+Thus our final scheme to consider is a variant of FallingEdge that also requires/assumes the maintenance of a _bumplog_, which is merely a degenerate changelog that contains nothing more than what's necessary to determine the next version number for each changed packaged (ie one classification of `patch`, `minor`, and `major` for each merged PR).
+
```

Comment:
Is `FallingEdge` like `FalingEdgePatch` but PR author doesn't bump the major version (thus regular PRs never bump versions at all)?